### DISCUSSION TITLE
Add IsMuted to MediaElement

### DISCUSSION THREAD

**Discussion by jamesmontemagno** [CommunityToolkit/Maui#950] [New Feature Discussions]
Right now we have to set values to 0 to mute and then have to remember the volume. Each media player has the capability to Mute. Would be nice to have a method to call with true/false ;)

**Comment by jfversluis** [comment]
Underway: https://github.com/CommunityToolkit/Maui/pull/953

**Reply by Kapusch** [reply]
Hey @jfversluis , thank you for the ShouldMute option!

I just had a quick question, is there a real need for which Volume should remain equal to 1 when ShouldMute is True?

I'm asking because I'm building my own controls on top of the MediaElement, and I have a Slider to control the Volume and a Mute Button. And **when User goes on mute, I have to set Volume to 0 so it reflects properly with the cursor of the attached Slider**.

Wouldn't it be better if we changed the behaviour of the MediaElement to set Volume to 0 when ShouldMute goes to True?

![mute-button-volume-tracker](https://github.com/CommunityToolkit/Maui/assets/9576733/3575cec7-9986-4831-9021-7278cc8d1b61)

**Reply by jfversluis** [reply]
No problem at all, the controls look great!

This implementation was done very intentional. I think mute and the volume are two different things. As a user, if I mute, I expect the volume to be restored when I unmute again. Same as my TV. If I mute the audio on there it goes silent, but if I unmute, the same level of volume is restored.

I guess this is a matter of preference, but I don't think we'll be changing it :)

**Reply by Kapusch** [reply]
Thanks!

Ok, I was thinking we could have internal variable where to store "volume in memory" to set back at same level when unmuting, but I'm either good keeping mute and volume isolated: this finally gives more flexibility for the developer.

Always a pleasure to discuss with you, keep going!

**Comment by vhugogarcia** [comment]
Closed per PR approved and merged. [https://github.com/CommunityToolkit/Maui/pull/953](https://github.com/CommunityToolkit/Maui/pull/953)
